tests/dynamic_data/users_roles/delete_user_test.py

```diff
@@ -1,47 +1,104 @@
 
         # check if publication info exists
         with app.app_context():
-            publication_info = layman_util.get_publication_info(USERNAME, publication_type, publication)
+            publication_info = layman_util.get_publication_info(username, publication_type, publication)
         assert isinstance(publication_info, dict) and publication_info, "Publication info cannot be empty"
 
     # check if workspace exists
-    assert workspace_exists(USERNAME), f"Workspace '{USERNAME}' was not found"
-    assert user_exists(USERNAME), f"User '{USERNAME}' was not found"
+    assert workspace_exists(username), f"Workspace '{username}' was not found"
+    # check if user exists
+    assert user_exists(username), f"User '{username}' was not found"
 
-    actor_name = USERNAME
-    response = process_client.delete_user(USERNAME, actor_name=actor_name)
+    response = process_client.delete_user(username, actor_name=actor_name)
     assert response.status_code == 200, response.json()
 
     # check if publication was deleted
     for publication_type in process_client.PUBLICATION_TYPES:
-        publications_after_delete = process_client.get_publications(publication_type, actor_name=USERNAME)
+        publications_after_delete = process_client.get_publications(publication_type, actor_name=username)
         assert not any(pub.get('name') == publication for pub in publications_after_delete), f"Publication {publications_after_delete} was not deleted"
         # check if publication info was deleted
         with app.app_context():
-            publication_info = layman_util.get_publication_info(USERNAME, publication_type, publication)
+            publication_info = layman_util.get_publication_info(username, publication_type, publication)
         assert isinstance(publication_info, dict) and not publication_info, "Publication info should be empty"
 
     # check if workspace was deleted
-    assert not workspace_exists(USERNAME), f"Workspace '{USERNAME}' was not deleted"
-    assert not user_exists(USERNAME), f"User '{USERNAME}' was not deleted"
+    assert not workspace_exists(username), f"Workspace '{username}' was not deleted"
+    # check if user was deleted
+    assert not user_exists(username), f"User '{username}' was not deleted"
+
+
+@pytest.mark.parametrize('setup_users_for_testing_errors', [
+    ('non_existing_user', 'test_delete_user_negative2', 404, 57),
+    ('test_delete_user_negative', 'test_delete_user_negative2', 403, 30),
+    ('', 'test_delete_user_negative2', 404, None),
+    (ANONYM_USER, 'test_delete_user_negative2', 404, 57),
+    (NONAME_USER, 'test_delete_user_negative2', 404, 57),
+], indirect=True)
+@pytest.mark.usefixtures('ensure_layman_module', 'oauth2_provider_mock')
+def test_delete_user_negative(setup_users_for_testing_errors):
+    username, actor_name, expected_status, layman_code = setup_users_for_testing_errors
```

Comment:
```suggestion
    username, actor_name, expected_status, exp_layman_code = setup_users_for_testing_errors
```

In tests, we usually use `exp_` prefix (exp = expected) for variables with expected values.